examples/src/bin/auto_composite_template.rs

```diff
@@ -0,0 +1,64 @@
+//! # Composite Template Example
+//!
+//! This sample demonstrates how to create a widget using GTK's composite templates.
+
+use gio::prelude::*;
+use glib::subclass::prelude::*;
+use glib::{glib_object_subclass, glib_wrapper};
+use gtk::gtk4_macros::*;
+use gtk::prelude::*;
+
+use gtk::subclass::prelude::*;
+use gtk::subclass::widget::*;
+
+mod imp {
+    use super::*;
+
+    /// The private struct, which will hold the automatically generated widgets
+    #[template(file = "examples/src/bin/composite_template.ui")]
+    #[derive(Debug)]
+    pub struct ExApplicationWindow {}
```

Comment:
I didn't look at the code macro code yet, but is it possible to have other fields in the struct and those would be left alone by the macro?

Reply:
In theory yes. In practice this does not work because the code currently generates a ::new() function which needs to specify all struct members. Still need to figure out a better solution to this.